internal/abci/utils_test.go

```diff
@@ -121,7 +121,7 @@ func (n *fakeNode) NextHeight() int64 {
 }
 
 func (n *fakeNode) WriteStates() {
-	_, _, err := n.db.WriteStates(n.NextHeight())
+	_, _, err := n.db.Begin().Commit(n.NextHeight())
```

Comment:
I don't think `db.Begin().Commit(...)` serves any purpose. If tests still pass with `n.WriteStates()` removed, this method should be deleted. If that breaks tests, we should investigate why.